product-service/src/functions/getProductsList/handler.ts

```diff
@@ -1,9 +1,55 @@
-import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
-import { formatJSONResponse } from '@libs/api-gateway';
-import productsData from '../../__mocks__/products'
+import { ValidatedEventAPIGatewayProxyEvent, formatJSONResponse } from '@libs/api-gateway';
 
-const getProductsList: ValidatedEventAPIGatewayProxyEvent<any> = async (event) => {
-  return formatJSONResponse(productsData);
+import { DynamoDBClient, ScanCommand } from '@aws-sdk/client-dynamodb';
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+import type { TProducts, TProductsStock } from 'src/types/product';
+import { convertToProductStockType, convertToProductType } from '@libs/convert';
+
+const client = new DynamoDBClient();
+const docClient = DynamoDBDocumentClient.from(client);
+
+docClient.middlewareStack;
+
+const productsTable = async () => {
+  const command = new ScanCommand({
+    TableName: process.env.PRODUCTS_TABLE,
+  });
+
+  const result = await docClient.send(command);
+  const products: TProducts = result.Items.map((item) => {
+    return convertToProductType(item);
+  });
+  return products;
+};
+
+const productsStockTable = async () => {
+  const command = new ScanCommand({
+    TableName: process.env.PRODUCTS_STOCK_TABLE,
+  });
+
+  const result = await docClient.send(command);
+  const products: TProductsStock = result.Items.map((item) => {
+    return convertToProductStockType(item);
+  });
+  return products;
+};
+
+const getProductsList: ValidatedEventAPIGatewayProxyEvent<any> = async () => {
+  const products = await productsTable();
+  const productsStock = await productsStockTable();
```

Comment:
you can avoid of scan all stock table, by search by product_id on each product level